src/testdrive/run.py

```diff
@@ -132,15 +128,13 @@ def main():
         for test, *test_args in source.next():
             id_ = builder.build(os.path.dirname(test))
             testimpl = os.path.join(basedir, test)
-            if not args.timeless:
-                start = timenow()
+            start = timenow()
             result = drive(testimpl, *test_args)
-            if not args.timeless:
-                end = timenow()
+            end = timenow()
             result['id'] = id_
-            if not args.timeless:
+            if 'timestamp' not in result:
```

Comment:
Doubt: This case `'timestamp' not in result` only would apply for for environment checks right?

Reply:
Maybe, we would have to check how they are implemented. In any case the logic now is "if a timestamp wasn't given by the test, we must add one".